packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendar.tsx

```diff
@@ -199,87 +205,186 @@ export const DatepickerCalendar: ComponentWithAs<'div', DatepickerCalendarProps>
+  //   if (!weeksToShow) {
+  //     // Slicing because grid contains extra 1 week in the front and in the back.
+  //     return grid.slice(1, grid.length - 1);
+  //   }
+  //   return grid;
+  // };
+
+  const gridOptions = {
+    selectedDate,
+    navigatedDate: gridNavigatedDate,
+    firstDayOfWeek,
+    firstWeekOfYear,
+    dateRangeType,
   };
 
-  const grid = getSlicedGrid();
+  const grid = getDayGrid(gridOptions); // getSlicedGrid();
+  const getInitialFocusDate = () => {
+    if (selectedDate) {
+      const normalizedSelectedDate = new Date(selectedDate.toString());
+      normalizedSelectedDate.setHours(0);
+      normalizedSelectedDate.setMinutes(0);
+      normalizedSelectedDate.setSeconds(0);
+      normalizedSelectedDate.setMilliseconds(0);
+      return normalizedSelectedDate;
+    }
+    const todayDate = _.flatten(grid).find(day => day.isToday);
+    return todayDate.originalDate;
+  };
+  const [focusedDate, setFocusedDate] = React.useState<Date>(getInitialFocusDate());
 
   const changeMonth = (nextMonth: boolean) => {
     const updatedGridNavigatedDate = addMonths(gridNavigatedDate, nextMonth ? 1 : -1);
     setGridNavigatedDate(updatedGridNavigatedDate);
+    setFocusedDate(updatedGridNavigatedDate);
   };
 
+  const itemRefs = React.useMemo<Map<string, React.RefObject<HTMLElement>>>(() => {
+    const refs = new Map<string, React.RefObject<HTMLElement>>();
+    _.map(grid, week => _.map(week, (day: IDay) => (refs[day.key] = React.createRef())));
+    return refs;
+  }, [grid]);
+
+  // const handleFocus = (day: IDay, hiddenCell: boolean): void => {
+  //   console.log(day, hiddenCell);
+  // };
+
+  const handleKeyDown = (e, day) => {
+    const keyCode = getCode(e);
+    const initialDate = day.originalDate;
+    let targetDate: Date | null = null;
+    let direction = 1; // by default search forward
+
+    switch (keyCode) {
+      case keyboardKey.ArrowDown: {
+        targetDate = addWeeks(initialDate, 1);
+        break;
+      }
+      case keyboardKey.ArrowUp: {
+        targetDate = addWeeks(initialDate, -1);
+        direction = -1;
+        break;
+      }
+      case keyboardKey.ArrowLeft: {
+        direction = -1;
+        targetDate = addDays(initialDate, -1);
+        break;
+      }
+      case keyboardKey.ArrowRight: {
+        targetDate = addDays(initialDate, 1);
+        break;
+      }
+      default:
+        break;
+    }
+
+    if (!targetDate) {
+      // if we couldn't find a target date at all, do nothing
+      return;
+    }
+    const newNavigateDate = findAvailableDate({
+      initialDate,
+      targetDate,
+      direction,
+    });
+    if (!dayInGrid(grid.slice(1, grid.length - 1), newNavigateDate)) {
+      setGridNavigatedDate(newNavigateDate);
+      setFocusedDate(newNavigateDate);
+      e.preventDefault();
+    }
+  };
+
+  React.useEffect(() => {
+    if (focusedDate) {
+      const cellToFocus = itemRefs[focusedDate.toString()];
```

Comment:
This looks a bit scary a it's platform dependent and can be an issue for SSR rendering

Reply:
day.key is computed as date.toString
I can create a method `getDateKey(date:Date) => date.toString()` in date-time-utilities packages and use it both in `getDayGrid` and here, to guarantee the consistency
WDYT @layershifter ?

Reply:
I understand that it might not solve SSR, so yeah, let's sync on that